### PR TITLE
Search feature

### PR DESCRIPTION
**Pull request by rajat98**

### CODE REVIEW

File: backend/data_models.py
Comment (Ztrimus) [inline backend/data_models.py:1]: - Make non-mandatory fields optional 
- you can apply validator like max length or szie

**Review by Ztrimus — COMMENTED** [review]
Good to go!

File: backend/database/DDL.sql
Comment (RohanAwhad) [inline backend/database/DDL.sql:1]: I trust you

File: backend/db.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        coalesce(1.0 / ({} + semantic.rank_ix), 0.0) * {}
+            desc
+    limit
+        least({}, 30)""".format(rrf_k, full_text_weight, rrf_k, semantic_weight, match_count)
+    search_query = (search_query_with_part + search_query_select_part + search_query_where_part +
+                    search_query_order_by_limit_part)
+
+    with conn.cursor() as cur:
+        cur.execute(sql.SQL(search_query))
+        result = cur.fetchone()
```
Comment (RohanAwhad) [inline backend/db.py]: should have been `fetchall()`

**Pull request merged**